libs/community/langchain_community/embeddings/huggingface.py

```diff
@@ -87,10 +92,24 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             pool = self.client.start_multi_process_pool()
             embeddings = self.client.encode_multi_process(texts, pool)
             sentence_transformers.SentenceTransformer.stop_multi_process_pool(pool)
-        else:
-            embeddings = self.client.encode(texts, **self.encode_kwargs)
 
-        return embeddings.tolist()
+            return embeddings.toList()
+        else:
+            if self.show_progress:
+                try:
+                    from tqdm import tqdm
+
+                    iter_ = tqdm(texts, desc="HuggingFaceEmbeddings")
+                except ImportError:
+                    logging.warning(
+                        "Unable to show progress because tqdm could not be imported."
+                        "Please install it with `pip install tqdm`."
+                    )
+                    iter_ = texts
+            else:
+                embeddings = self.client.encode(texts, **self.encode_kwargs)
+
+        return [self.client.encode(text, **self.encode_kwargs) for text in iter_]
```

Comment:
do we know if this affects speed?

Reply:
probably fine either way since this is opt-in, but worth documenting if it does slow things down

Reply:
I've seen the same format in other community embedding functions who have implemented a progress bar.

If we look at the encoding function that gets called in [`sentence-transformers`](https://github.com/UKPLab/sentence-transformers/blob/2c4d21cfd5d6bb7bf6067619a701320537d29a01/sentence_transformers/SentenceTransformer.py#L204-L262) it has handling for both strings and lists of strings, so this would allow us to skip the sorting steps that occur in the function that would otherwise just be overhead for a single string.

Reply:
The underlying API implements batch optimization based on the referenced code. We should not be sacrificing that even as an opt-in with show_progress=bar. Users will not know and the performance loss is not worth it.

Instead we can use the `show_progress` of the underlying encode function.